### DISCUSSION TITLE
Troubleshooting Linux instance !!!

### DISCUSSION THREAD

**Discussion by joindevopscloud** [daws-76s/.github#2] [Q&A]
When you are unable to login to a Linux machine, what are the interrogative steps that you follow to troubleshoot the issue?

**Comment by devopsden2020** [comment]
you need to check the .pem file path and the public IP from the AWS instance

**Reply by cloudtechprem** [reply]
1. Check the Security Groups (Fire Wall) of the Linux machine, If you are connecting through SSH, make sure port 22 is enabled.
2. check the IP address that you are trying to connect (Public IP or Private IP)
3. If you are trying to connect with the Username and Password, make sure you use the correct ones.
4. Check the .pem file.

**Comment by knbob** [comment]
1. Check the security group in which it should allow the connections.
2. Check the putty/super putty configurations if any.
3. Check the username and password.
4. Check the private key path if key based authentication instance created.
5. Make sure select proceed without key pair or key pair based on your requirement.
6. Make sure public address to be used for logging ec2 instance.

**Comment by saipdevops555** [comment]
**From network Side - Unable to connect to Linux machine :**

1. Start by verifying the physical connections if you have access to them.
2. Proceed to examining the network configuration using commands like ifconfig or ip addr. 
3. Check if the network interface is up and has the correct internet protocol (IP) address. 
4. Next, test the connectivity to the local network with ping and inspect routing with route -n or ip route. 
5. Verify the domain name system (DNS) configuration in /etc/resolv.conf and test DNS resolution. 
6. If a firewall is present, review the rules to ensure it's not blocking the necessary traffic. 
7. Analyze the output of the netstat command to reveal potential issues with listening ports. 
8. Lastly, review system and network logs found in /var/log, which might give clues to specific issues. 

EC2 Instance Issue :: 
1) Check the security group for this EC2 instance. You may have set it (as you should) to allow SSH from "My IP Address". But your IP address may have changed if you reset your router, etc. You may need to correct the IP Address that is allowed to connect.
2) In the AWS management console and examine the EC2 instance. It will tell you the name of the key file selected for that instance. Confirm that it is the file you expect it to be. Make sure your SSH configuration is using the correct private key file to match that public key file.
3) Check the instance's public IP address and/or public URL. If you applied an Elastic IP, both of these would have changed. You may need to correct the host name in your SSH connection.
4) Check your Putty configurations and the connections and authentications used( if right PPK has been used)
